test/common.js

```diff
@@ -37,9 +46,9 @@ if (process.env.NODE_COMMON_PIPE) {
   }
 }
 
-try {
-  fs.accessSync(exports.opensslCli);
-} catch (err) {
+var openssl_cmd = child_process.spawnSync(exports.opensslCli, ['version']);
```

Comment:
Ideally, this only runs when the openssl binary is actually needed.  Spawning a child process adds a few milliseconds to each test; multiplied by ~600 tests, that's a couple of extra seconds.  Maybe you can turn the `.opensslCli` property into a lazy getter?